### PR TITLE
Week9

### PR DESCRIPTION
**Pull request by ruhankhandakar**

### CODE REVIEW

**Comment by ShashankSuresh** [comment]
@ruhankhandakar `Week-9/todo/src/components/index.jsx` - this is an empty file, do you need this?

File: Week-9/todo/src/context/index.js [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+    const selectedNoteId = state.noteInputData.id || nodeId;
+    let allUpdatedNotes = state.allNotes.filter(
+      (note) => note.id !== selectedNoteId
+    );
+
+    dispatch({
+      type: UPDATE_ALL_NOTES,
+      payload: allUpdatedNotes,
+    });
+    console.log("allUpdatedNotes", allUpdatedNotes);
```
Comment (ShashankSuresh) [inline Week-9/todo/src/context/index.js]: Please remove `console.log` statement.

File: Week-9/todo/src/context/index.js [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+    });
+    dispatch({
+      type: TOGGLE_COMPLETION,
+      payload: allUpdatedNotes,
+    });
+    storeInLocalStorage("notes", allUpdatedNotes);
+  };
+
+  const editNote = (updatedNote) => {
+    console.log("updatedNote", updatedNote);
```
Comment (ShashankSuresh) [inline Week-9/todo/src/context/index.js]: Please remove `console.log` statement.

File: Week-9/todo/src/context/index.js [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+        }
+        if (updatedNote.note) {
+          note.note = updatedNote.note;
+        }
+
+        note.updated = new Date().getTime();
+      }
+      return note;
+    });
+    console.log("allUpdatedNotes", allUpdatedNotes);
```
Comment (ShashankSuresh) [inline Week-9/todo/src/context/index.js]: Please remove `console.log` statement.

File: Week-9/todo/src/serviceWorkerRegistration.js [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+        if (installingWorker == null) {
+          return;
+        }
+        installingWorker.onstatechange = () => {
+          if (installingWorker.state === "installed") {
+            if (navigator.serviceWorker.controller) {
+              // At this point, the updated precached content has been fetched,
+              // but the previous service worker will still serve the older
+              // content until all client tabs are closed.
+              console.log(
```
Comment (ShashankSuresh) [inline Week-9/todo/src/serviceWorkerRegistration.js]: Please remove `console.log` statement.

File: Week-9/todo/src/serviceWorkerRegistration.js [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+
+              // Execute callback
+              if (config && config.onUpdate) {
+                config.onUpdate(registration);
+              }
+            } else {
+              // At this point, everything has been precached.
+              // It's the perfect time to display a
+              // "Content is cached for offline use." message.
+              console.log("Content is cached for offline use.");
```
Comment (ShashankSuresh) [inline Week-9/todo/src/serviceWorkerRegistration.js]: Please remove `console.log` statement.

**Comment by ShashankSuresh** [comment]
Im able to add an empty Note. We will have to handle this condition. 

Also, can we have 2 notes with the same Title?

image

**Comment by ruhankhandakar** [comment]
@ShashankSuresh fixed the empty case. and for the duplicate -> I tried to copy google keeps, and there they support duplicate notes. If you want I'll add duplicate handling.

**Review by ShashankSuresh — CHANGES_REQUESTED** [review]

**Pull request merged**